Added "Continue" link to error page (#40909)

mijarena opened the pull request:
<!--
Thanks for submitting a change to ownCloud!

This is the bug tracker for the Server component. Find other components at https://github.com/owncloud/core/blob/master/.github/CONTRIBUTING.md#guidelines

For fixing potential security issues please see https://owncloud.com/security/

To make it possible for us to get your change reviewed and merged please carefully fill out the requested information below.

Please note that any kind of change needs first be submitted to the master branch which holds the next version of ownCloud.

Please set the following labels:

- Set label "3 - To review" for review or "2 - Development" if the PR still has open tasks
- Assignment: assign to self
- Milestone: set the same as the ticket this PR fixes, or "development" by default
- Reviewers: pick at least one
-->

## Description
<!--- Describe your changes in detail -->

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
- #40909

## Motivation and Context
User can quickly return to the starting page when encountering an error.


## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
- test environment:
- test case 1:
- test case 2:
- ...

## Screenshots (if appropriate):
![image](https://github.com/owncloud/core/assets/85600105/02eb58ce-91b9-421b-961b-16dcec7e029d)


## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
<!-- Tick the checkboxes when done. -->
<!-- Raise documentation ticket in https://github.com/owncloud/documentation -->
- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

mijarena committed:
Added "Continue" link to error page (#40909)

User can quickly return to the starting page when encountering an error.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/owncloud/core?pullRequest=40910) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/owncloud/core?pullRequest=40910) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/owncloud/core?pullRequest=40910) it.</sub>

mijarena committed:
Fix formatting

jnweiger reviewed (changes requested):
href="/" is not always correct.

There are two config variables that need to be taken into account:
```
/**
 * Define the Web base URL
 * This key is necessary for the navigation item to the new ownCloud Web UI and for redirecting
 * public and private links.
 */
'web.baseUrl' => '',
/**
 * Define clean URLs without `/index.php`
 * This parameter will be written as `RewriteBase` on update and installation of
 * ownCloud to your `.htaccess` file. While this value is often simply the URL
 * path of the ownCloud installation it cannot be set automatically properly in
 * every scenario and needs thus some manual configuration.
 *
 * In a standard Apache setup this usually equals the folder that ownCloud is
 * accessible at. So if ownCloud is accessible via `https://mycloud.org/owncloud`
 * the correct value would most likely be `/owncloud`. If ownCloud is running
 * under `https://mycloud.org/` then it would be `/`.
 *
 * Note that the above rule is not valid in every case, as there are some rare setup
 * cases where this may not apply. However, to avoid any update problems this
 * configuration value is explicitly opt-in.
 *
 * After setting this value run `{occ-command-example-prefix} maintenance:update:htaccess`. Now, when the
 * following conditions are met ownCloud URLs won't contain `index.php`:
 *
 * - `mod_rewrite` is installed
 * - `mod_env` is installed
 */
'htaccess.RewriteBase' => '/',
```

ownCloud may also be installed like this:

https://oc10130a1-20230807.jw-qa.owncloud.works/mammamia/owncloud/login
config.php: `'htaccess.RewriteBase' => '/mammamia/owncloud',`

https://oc1012-index-20230807.jw-qa.owncloud.works/index.php
No RewriteBase in config.php

https://oc10122-both-20230807.jw-qa.owncloud.works/big-bang/index.php
apache.conf: `Alias /big-bang "/var/www/owncloud/"` without a RewriteBase in config.php


https://oc10130a1-web-app-20230807.jw-qa.owncloud.works/index.php
... and what happens with the new web UI? (Optional, but it should not break...)

mijarena committed:
Fix at base url for continue-link

mijarena committed:
User can quickly return to the starting page when encountering an error.

Now not only at the error page (template error.php) but also at the 403 page (template 403.php)

IL10N is not working for "Continue", but as part of the message is in english anyway, this is a fix for later.

mijarena commented:
Fixed on 403 page, too

jnweiger reviewed (approved):
Work in all my test cases.
* admin permission, wrong file id
* index.php less and other
* / and subfolder install.

IljaN commented:
- Should we make the "Continue" prettier. Maybe under the message box? But fine for me.
- CI is failing https://drone.owncloud.com/owncloud/core/38908/3/7
- Please add a changelog entry https://github.com/owncloud/core/pull/40910#issuecomment-1667830427
- Added additional reviewers since we worked on the code together ;)

IljaN reviewed (changes requested):
- Should we make the "Continue" prettier. Maybe under the message box? But fine for me.
- CI is failing https://drone.owncloud.com/owncloud/core/38908/3/7
- Please add a changelog entry https://github.com/owncloud/core/pull/40910#issuecomment-1667830427
- Added additional reviewers since we worked on the code together ;)

mijarena commented:
I made this to be working correctly So it could be an accepted feature for the next release. I planed to polish code, style and translation next during beta.

jvillafanez commented on apps/files/lib/Controller/ViewController.php:
```diff
@@ -335,7 +335,17 @@ public function showFile($fileId, $details = null) {
+				'error',
+				[
+					"errors" => [$param],
+					"base_url" => \OC::$server->getURLGenerator()->getAbsoluteURL("/"),
```
Is it possible to inject the `URLGenerator`?

jvillafanez reviewed (commented):
Not sure if it's easy to add some basic unit tests.

DeepDiver1975 commented:
cleaning up - I think this is no longer of anybody's interest
